sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/profile/AzureProfile.java

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.management.resources.fluentcore.profile;
+
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.util.Configuration;
+
+/**
+ * Azure profile for client.
+ */
+public class AzureProfile {
```

Comment:
Does it need `withEnvironment`?

Reply:
Hmmm looks withEnvironment can be removed after I checked. The environment is prerequisite before creating `Authenticated` instance.

Reply:
Should we use a Builder to make this class immutable? Do we expect user be able to change `subscriptionId` or `tenantId` in the middle of running the code.

Reply:
The user might want to set `subscriptionId` or `tenantId` after listing subs/tenants via `Authenticated`. Inside each manager, we just keep the `subscriptionId` and `environment` instead of profile. Profile is only used in the constructor of manager.

Reply:
That is method inside AzureConfigure, which does not related to immutable or immutable of this class. You can always create a new AzureProfile there, like what you did to `HttpPipeline`, which is immutable.